support-models/src/main/scala/com/gu/support/workers/states/CreateZuoraSubscriptionState.scala

```diff
@@ -1,31 +1,100 @@
 package com.gu.support.workers.states
 
-import java.util.UUID
+import com.gu.i18n.Country
 
-import com.gu.salesforce.Salesforce.SalesforceContactRecords
+import java.util.UUID
+import com.gu.salesforce.Salesforce.{SalesforceContactRecords, SfContactId}
+import com.gu.support.encoding.Codec.deriveCodec
 import com.gu.support.promotions.PromoCode
 import com.gu.support.workers.{PaymentMethod, SalesforceContactRecord, User, _}
 import org.joda.time.LocalDate
-import com.gu.support.encoding.CustomCodecs.{decodeLocalTime, encodeLocalTime}
+import com.gu.support.encoding.CustomCodecs.{decodeLocalTime, encodeLocalTime, decodeCountry, encodeCountryAsAlpha2}
+import com.gu.support.encoding.{Codec, DiscriminatedType}
 import com.gu.support.redemptions.RedemptionData
+import com.gu.support.workers.GiftRecipient.{DigitalSubscriptionGiftRecipient, WeeklyGiftRecipient}
 
-case class CreateZuoraSubscriptionState(
+case class CreateZuoraSubscriptionWrapperState(
```

Comment:
the old state that covered everything optionally was called `CreateZuoraSubscriptionState`

Now that it's split into "state that you always have", and "product specific state", I have called them `CreateZuoraSubscriptionWrapperState` and `CreateZuoraSubscriptionState` respectively.  It is more of a minimal change.

However it would make more sense to keep the original name for the "wrapper" and call the product specific state `CreateZuoraSubscriptionProductSpecificState`

Thoughts about naming (or simplifying it conceptially) would be nice.  I couldn't really flatten it into a single level and retain the simplicity of the second level.

Reply:
I think I'd rather keep `CreateZuoraSubscriptionState` as the outer class as it maps to the lambda name and then call the inner class something like `CreateZuoraSubscriptionProductState`.

Reply:
yeah that does sound like a better way, will do that